corehq/apps/users/models.py

```diff
@@ -2440,6 +2440,8 @@ def add_domain_membership(self, domain, timezone=None, **kwargs):
         super().add_domain_membership(domain, timezone, **kwargs)
 
     def delete_domain_membership(self, domain, create_record=False):
+        self._leaving_domains = getattr(self, '_leaving_domains', []) + [domain]
```

Comment:
I could have also just called `sync_web_usercases_if_applicable` here, but that didn't feel quite right, since it's already called in `save`, and at this point, removing the domain membership hasn't been committed.

Reply:
Yeah, ever since moving user data to SQL I've been cognizant of how much of what's stored on the user docs is conceptually a related model.  It'll be tedious business to tease this all apart when we move it to SQL, but this would hopefully be so much easier then.  Save changes to domain memberships separately from changes to demographic information (and certainly separately from things like `login_attempts` or `last_submission_for_user`).  If we do need to group things together, we can use a normal transaction :exploding_head: